crates/ruff_python_formatter/src/string/normalize.rs

```diff
@@ -913,11 +939,46 @@ pub(super) fn is_fstring_with_quoted_debug_expression(
         } else {
             false
         }
-    }) {
-        return true;
-    }
+    })
+}
 
-    false
+/// Returns `true` if `string` is an f-string part that contains a debug expression with a format spec
+/// that contains the opposite quote. It's important to preserve the quote style for those f-strings
+/// because changing the quote style would result in invalid syntax.
+///
+/// ```python
+/// f'{1=: "abcd \'\'}'
+/// ```
+pub(super) fn is_fstring_with_quoted_format_spec_and_debug(
+    fstring: &FString,
```

Comment:
To be consistent, we're using this format everywhere although I see it's not being enforced and there are usages of both, I can change that in a different PR
```suggestion
    f_string: &FString,
```

Reply:
I prefer to keept it as is so that it's consistent with the function name, but I agree that we have both forms of spelling fstring. Although I don't think this is a big deal